wgpu-core/src/id.rs

```diff
@@ -1,9 +1,27 @@
 use crate::{Epoch, Index};
-use std::{cmp::Ordering, fmt, marker::PhantomData, num::NonZeroU64};
+#[cfg(feature = "id32")]
+use std::num::NonZeroU32;
+#[cfg(not(feature = "id32"))]
+use std::num::NonZeroU64;
+use std::{cmp::Ordering, fmt, marker::PhantomData};
 use wgt::Backend;
 
+#[cfg(feature = "id32")]
+type IdType = u32;
+#[cfg(not(feature = "id32"))]
+type IdType = u64;
+#[cfg(feature = "id32")]
+type ZippedIndexType = u16;
+#[cfg(not(feature = "id32"))]
+type ZippedIndexType = Index;
+#[cfg(feature = "id32")]
+const INDEX_BITS: usize = 16;
+#[cfg(not(feature = "id32"))]
+const INDEX_BITS: usize = 32;
```

Comment:
we can use (size_of)[https://doc.rust-lang.org/stable/std/mem/fn.size_of.html] for all the places instead of INDEX_BITS: `size_of::<IdType>()`. It's a const fn, so it can be used everywhere.